pkg/console/console.go

```diff
@@ -58,11 +65,41 @@ func Run() error {
 	restful.Filter(cors.Filter)
 	restful.Filter(restful.OPTIONSFilter())
 
+	tlsConfigWatch := tlsconfig.NewWatch(
+		path.Join(configDir, TlsProfileFile),
+		serviceCertPath,
+		serviceKeyPath,
+	)
+	tlsConfigWatch.Reload()
+
+	watchDone := make(chan struct{})
+	defer close(watchDone)
+	go func() {
+		err := tlsConfigWatch.Run(watchDone)
+		log.Log.Errorf("Error running TLS config watch: %s", err)
+	}()
+
 	server := &http.Server{
 		Addr: fmt.Sprintf("%s:%d", defaultAddress, defaultPort),
+		TLSConfig: &tls.Config{
+			GetConfigForClient: func(_ *tls.ClientHelloInfo) (*tls.Config, error) {
+				config, err := tlsConfigWatch.GetConfig()
+				if err != nil {
+					return nil, err
+				}
+				config.Certificates = []tls.Certificate{*serviceCert}
```

Comment:
This is using the static `serviceCert` too. Can you make sure it is reloaded on changes to the certificate?

Reply:
Also why don't you use the `GetCertificate` func below to return this certificate instead of writing it into `config.Certificates`?

Reply:
This was a bug, thanks for noticing.

I don't exactly understand what do you mean about the `GetCertificate()`.
The server behaves like this: When TLS handshake is started, it calls `config.GetConfigForClient()` function on it's config to get a new config. And then this new config is used for the rest of the connection. So any configuration in the old config is ignored, including the `GetCertificate()` function that is set below.

Reply:
Okay, thanks for the clarification.